File: Python Excel Examples/WorkbookApiDemo/unProtectDocument.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.workbook_api.WorkbookApi(configuration)

name = "UnProtectDocument.xlsx"
storage = ""
folder = "ExcelDocument"
protection = spirecloudexcel.models.WorkbookProtectionRequest()
protection.password = "123"
protection.protection_type = "READONLY"  # Available type: ALL, READONLY, STRUCTURE, WINDOWS
api.un_protect_document(name, protection, folder=folder, storage=storage)

File: PHP Excel Examples/WorkbookApiDemo/unProtectDocument.php
<?php

use Spire\Cloud\Excel\Sdk\Configuration;
use Spire\Cloud\Excel\Sdk\Model\WorkbookProtectionRequest;

$appId = "your id";
$appKey = "your key";
$baseUrl = "https://api.e-iceblue.cn";
$configuration = new Configuration($appId, $appKey, $baseUrl);
$apiInstance = new WorkbookApi($configuration);

$name = "UnProtectDocument.xlsx";
$storage = null;
$folder = "input";
$encryption = new WorkbookProtectionRequest();
$encryption->setPassword("123");
#supports types: ALL, READONLY, STRUCTURE, WINDOWS
$encryption->setProtectionType("READONLY");
$result = $apiInstance->UnProtectDocument($name, $encryption, $folder, $storage);

